test/new-e2e/tests/windows/agent/agent.go

```diff
@@ -24,3 +42,48 @@ func UninstallAgent(host *components.RemoteHost, logPath string) error {
 	}
 	return windows.UninstallMSI(host, product, logPath)
 }
+
+// HasValidDatadogCodeSignature an error if the file at the given path is not validy signed by the Datadog Code Signing certificate
+func HasValidDatadogCodeSignature(host *components.RemoteHost, path string) error {
+	sig, err := windows.GetAuthenticodeSignature(host, path)
+	if err != nil {
+		return err
+	}
+	if !sig.Valid() {
+		return fmt.Errorf("signature status is not valid: %s", sig.StatusMessage)
+	}
+	if !strings.EqualFold(sig.SignerCertificate.Thumbprint, DatadogCodeSignatureThumbprint) {
+		return fmt.Errorf("signature thumbprint is not valid: %s", sig.SignerCertificate.Thumbprint)
+	}
+	return nil
+}
+
+// TestValidDatadogCodeSignatures verifies that the files at the given paths are validly signed by the Datadog Code Signing certificate
+// This test is skipped if the verify_code_signature parameter is set to false.
+func TestValidDatadogCodeSignatures(t *testing.T, host *components.RemoteHost, paths []string) bool {
+	return t.Run("code signatures", func(t *testing.T) {
```

Comment:
<!-- Suggestions propose improvements to the current subject. It’s important to be explicit and clear on what is being suggested and why it is an improvement. -->
**💬 suggestion**
Recommend marking testing helpers to improve test logs readability - if this fails, would report it as the caller address. This is a good article about `t.Helper` usage https://www.alexedwards.net/blog/easy-test-assertions-with-go-generics
```suggestion
func TestValidDatadogCodeSignatures(t *testing.T, host *components.RemoteHost, paths []string) bool {
         t.Helper()
	return t.Run("code signatures", func(t *testing.T) {
```